From 255b93e4222ca9f50d028dc9b940e80b8b2b0b1a Mon Sep 17 00:00:00 2001
From: Aleksandr Korzun <a.korzun@partner.samsung.com>
Date: Wed, 12 Jul 2023 13:27:06 +0300
Subject: [PATCH 127/128] [sls] Adapt to new SLS API header paths.

SAIT/PNMLibrary#566 renamed PNM API header directory from `pnmlib/ai`
to `pnmlib/sls`. This commit makes Pytorch use the new names.

References: MCS23-907

Signed-off-by: Aleksandr Korzun <a.korzun@partner.samsung.com>
---
 aten/src/ATen/native/pnm/PNMSecureTensor.h     | 2 +-
 aten/src/ATen/native/pnm/PNMTensor.h           | 6 +++---
 aten/src/ATen/native/pnm/ops/EmbeddingBag.cpp  | 4 ++--
 caffe2/operators/lengths_reducer_ops_pnm.cc    | 6 +++---
 caffe2/operators/lengths_reducer_ops_sec_pnm.h | 2 +-
 caffe2/python/pybind_state_pnm.cc              | 4 ++--
 6 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/aten/src/ATen/native/pnm/PNMSecureTensor.h b/aten/src/ATen/native/pnm/PNMSecureTensor.h
index d64d654e..1c84cb2c 100644
--- a/aten/src/ATen/native/pnm/PNMSecureTensor.h
+++ b/aten/src/ATen/native/pnm/PNMSecureTensor.h
@@ -1,8 +1,8 @@
 #ifndef __PNM_SECURE_TENSOR__
 #define __PNM_SECURE_TENSOR__
 
-#include <pnmlib/ai/sls.h>
 #include <pnmlib/secure/base_runner.h>
+#include <pnmlib/sls/sls.h>
 
 #include <ATen/ATen.h>
 #include <ATen/native/pnm/PNMBaseTensor.h>
diff --git a/aten/src/ATen/native/pnm/PNMTensor.h b/aten/src/ATen/native/pnm/PNMTensor.h
index b5df3f46..5e292d14 100644
--- a/aten/src/ATen/native/pnm/PNMTensor.h
+++ b/aten/src/ATen/native/pnm/PNMTensor.h
@@ -1,9 +1,9 @@
 #ifndef __PNM_TENSOR__
 #define __PNM_TENSOR__
 
-#include <pnmlib/ai/embedded_tables.h>
-#include <pnmlib/ai/sls.h>
-#include <pnmlib/ai/sls_type.h>
+#include <pnmlib/sls/embedded_tables.h>
+#include <pnmlib/sls/sls.h>
+#include <pnmlib/sls/type.h>
 
 #include <pnmlib/core/runner.h>
 #include <pnmlib/core/sls_device.h>
diff --git a/aten/src/ATen/native/pnm/ops/EmbeddingBag.cpp b/aten/src/ATen/native/pnm/ops/EmbeddingBag.cpp
index 223c64ca..c99eaaaa 100644
--- a/aten/src/ATen/native/pnm/ops/EmbeddingBag.cpp
+++ b/aten/src/ATen/native/pnm/ops/EmbeddingBag.cpp
@@ -1,5 +1,5 @@
-#include <pnmlib/ai/sls.h>
-#include <pnmlib/ai/sls_operation.h>
+#include <pnmlib/sls/operation.h>
+#include <pnmlib/sls/sls.h>
 
 #include <ATen/native/pnm/PNMTensorConvert.h>
 
diff --git a/caffe2/operators/lengths_reducer_ops_pnm.cc b/caffe2/operators/lengths_reducer_ops_pnm.cc
index d3d70d97..f357414e 100644
--- a/caffe2/operators/lengths_reducer_ops_pnm.cc
+++ b/caffe2/operators/lengths_reducer_ops_pnm.cc
@@ -5,13 +5,13 @@
 #include "caffe2/operators/segment_reduction_op.h"
 #include "caffe2/utils/math.h"
 
-#include <pnmlib/ai/embedded_tables.h>
-#include <pnmlib/ai/sls.h>
-#include <pnmlib/ai/sls_operation.h>
 #include <pnmlib/common/error.h>
 #include <pnmlib/common/views.h>
 #include <pnmlib/core/runner.h>
 #include <pnmlib/core/sls_device.h>
+#include <pnmlib/sls/embedded_tables.h>
+#include <pnmlib/sls/operation.h>
+#include <pnmlib/sls/sls.h>
 
 #include <algorithm>
 #include <vector>
diff --git a/caffe2/operators/lengths_reducer_ops_sec_pnm.h b/caffe2/operators/lengths_reducer_ops_sec_pnm.h
index 02068987..66ca2572 100644
--- a/caffe2/operators/lengths_reducer_ops_sec_pnm.h
+++ b/caffe2/operators/lengths_reducer_ops_sec_pnm.h
@@ -12,7 +12,7 @@
 #include <pnmlib/secure/sls_runner_factory.h>
 #include <pnmlib/secure/untrusted_sls_params.h>
 
-#include <pnmlib/ai/sls.h>
+#include <pnmlib/sls/sls.h>
 
 #include <pnmlib/common/128bit_math.h>
 
diff --git a/caffe2/python/pybind_state_pnm.cc b/caffe2/python/pybind_state_pnm.cc
index 4c6fbce2..5fff1df7 100644
--- a/caffe2/python/pybind_state_pnm.cc
+++ b/caffe2/python/pybind_state_pnm.cc
@@ -14,9 +14,9 @@
 #include "caffe2/operators/operator_fallback_pnm.h"
 #include "caffe2/python/pybind_state_registry.h"
 
-#include <pnmlib/ai/sls.h>
-#include <pnmlib/ai/embedded_tables.h>
 #include <pnmlib/core/sls_device.h>
+#include <pnmlib/sls/embedded_tables.h>
+#include <pnmlib/sls/sls.h>
 
 #include <fcntl.h>
 #include <sys/mman.h>
-- 
2.34.1

